### ISSUE TITLE
Trouble compiling GregSpray

### ISSUE THREAD

**Issue by ngoldbaum** [TACC/GraviT#11]
I'm trying to follow the instructions for building GraviT on the `dev` branch, but am hitting issues while trying to compile `GregSpray`:

My cmake invocation looks like:

```
embree_DIR=../embree cmake -DCMAKE_PREFIX_PATH=../ispc/ispc-v1.9.2-osx -DCMAKE_INSTALL_PREFIX:PATH=$PWD/install -DOSPRAY_MODULE_OPENGL_UTIL=True -DOSPRAY_USE_EXTERNAL_EMBREE=True .
```

And here's the output I see: https://gist.github.com/ngoldbaum/49043a3aa62648a9976f87a7d5e3cfcd

The error text is:

```
CMake Error at cmake/ospray_macros.cmake:168 (ADD_SUBDIRECTORY):
  ADD_SUBDIRECTORY given source "modules" which is not an existing directory.
Call Stack (most recent call first):
CMakeLists.txt:65 (OSPRAY_ADD_SUBDIRECTORY)
```

I tried creating a `modules` directory in the root of `GregSpray` but that led to the following error:

```
CMake Error at cmake/ospray_macros.cmake:168 (ADD_SUBDIRECTORY):
  The source directory

    /Users/goldbaum/Documents/GraviT/third-party/GregSpray/modules

  does not contain a CMakeLists.txt file.
Call Stack (most recent call first):
  CMakeLists.txt:65 (OSPRAY_ADD_SUBDIRECTORY)
```

Unfortunately I'm pretty unfamiliar with cmake so I'm not sure where to go from here.

**Comment by jgbarbosa** [comment]
Do you know which branch and/or hash you are using?

Sent from my iPhone

On Dec 8, 2017, at 17:09, Nathan Goldbaum <notifications@github.com<mailto:notifications@github.com>> wrote:


I'm trying to follow the instructions for building GraviT on the dev branch, but am hitting issues while trying to compile GregSpray:

My cmake invocation looks like:

embree_DIR=../embree cmake -DCMAKE_PREFIX_PATH=../ispc/ispc-v1.9.2-osx -DCMAKE_INSTALL_PREFIX:PATH=$PWD/install -DOSPRAY_MODULE_OPENGL_UTIL=True -DOSPRAY_USE_EXTERNAL_EMBREE=True .


And here's the output I see: https://gist.github.com/ngoldbaum/49043a3aa62648a9976f87a7d5e3cfcd

The error text is:

CMake Error at cmake/ospray_macros.cmake:168 (ADD_SUBDIRECTORY):
  ADD_SUBDIRECTORY given source "modules" which is not an existing directory.
Call Stack (most recent call first):
CMakeLists.txt:65 (OSPRAY_ADD_SUBDIRECTORY)


I tried creating a modules directory in the root of GregSpray but that led to the following error:

CMake Error at cmake/ospray_macros.cmake:168 (ADD_SUBDIRECTORY):
  The source directory

    /Users/goldbaum/Documents/GraviT/third-party/GregSpray/modules

  does not contain a CMakeLists.txt file.
Call Stack (most recent call first):
  CMakeLists.txt:65 (OSPRAY_ADD_SUBDIRECTORY)


Unfortunately I'm pretty unfamiliar with cmake so I'm not sure where to go from here.

**Comment by ngoldbaum** [comment]
The latest `dev` branch for GraviT, a300064a4.

**Comment by jgbarbosa** [comment]
Try this hash: 769f049


Joao Barbosa
Visualization Group
Texas Advanced Computing Center
University of Texas at Austin

On Dec 8, 2017, at 8:51 PM, Nathan Goldbaum <notifications@github.com<mailto:notifications@github.com>> wrote:


The latest dev branch for GraviT, a300064<https://github.com/TACC/GraviT/commit/a300064a4c038a32657ba0929f2038abdbdcde00>.

**Comment by ngoldbaum** [comment]
`error: pathspec '769f049' did not match any file(s) known to git.`

I don't think it's been pushed to github.

**Comment by pnav** [comment]
That’s the right hash, Nathan. I don’t see the hash Joao referenced, please stay at a300064 and will update ASAP with what I find.


On Dec 8, 2017, at 2:53 PM, Joao Barbosa <notifications@github.com<mailto:notifications@github.com>> wrote:

Try this hash: 769f049


Joao Barbosa
Visualization Group
Texas Advanced Computing Center
University of Texas at Austin

On Dec 8, 2017, at 8:51 PM, Nathan Goldbaum <notifications@github.com<mailto:notifications@github.com><mailto:notifications@github.com>> wrote:


The latest dev branch for GraviT, a300064<https://github.com/TACC/GraviT/commit/a300064a4c038a32657ba0929f2038abdbdcde00>.

**Comment by pnav** [comment]
Hi Nathan,

Sorry about that, the GregSpray/modules directory was inadvertently excluded from the checkin. That has been resolved in dev at the latest push (sha 761ee5c2).

Please pull and re-cmake on GregSpray.

Paul


On Dec 8, 2017, at 2:57 PM, Paul Navratil <pnav@tacc.utexas.edu<mailto:pnav@tacc.utexas.edu>> wrote:

That’s the right hash, Nathan. I don’t see the hash Joao referenced, please stay at a300064 and will update ASAP with what I find.


On Dec 8, 2017, at 2:53 PM, Joao Barbosa <notifications@github.com<mailto:notifications@github.com>> wrote:

Try this hash: 769f049


Joao Barbosa
Visualization Group
Texas Advanced Computing Center
University of Texas at Austin

On Dec 8, 2017, at 8:51 PM, Nathan Goldbaum <notifications@github.com<mailto:notifications@github.com><mailto:notifications@github.com>> wrote:


The latest dev branch for GraviT, a300064<https://github.com/TACC/GraviT/commit/a300064a4c038a32657ba0929f2038abdbdcde00>.

**Comment by ngoldbaum** [comment]
That fixed it! Thanks for the quick response :)
